cli/resource.go

```diff
@@ -24,7 +26,24 @@ func runResource(args *docopt.Args, client *controller.Client) error {
 	if args.Bool["add"] {
 		return runResourceAdd(args, client)
 	}
-	return fmt.Errorf("Top-level command not implemented.")
+
+	resources, err := client.AppResourceList(mustApp())
+	if err != nil {
+		return err
+	}
+
+	w := tabWriter()
+	defer w.Flush()
+
+	var provider *ct.Provider
+
+	listRec(w, "ID", "Provider ID", "Provider Name")
+	for _, j := range resources {
+		provider, err = client.GetProvider(j.ProviderID)
```

Comment:
Should CLI commands halt on first error? Asking earnestly. I wonder if it's
better to output as much as you can in a streaming format. Perhaps tabular
output to stdout, errors to stderr.

On Monday, March 2, 2015, Jonathan Rudenberg <notifications@github.com>
wrote:

> In cli/resource.go
> <https://github.com/flynn/flynn/pull/1137#discussion_r25658959>:
>
> > @@ -24,7 +26,24 @@ func runResource(args *docopt.Args, client *controller.Client) error {
> >  	if args.Bool["add"] {
> >  		return runResourceAdd(args, client)
> >  	}
> > -	return fmt.Errorf("Top-level command not implemented.")
> > +
> > +	resources, err := client.AppResourceList(mustApp())
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	w := tabWriter()
> > +	defer w.Flush()
> > +
> > +	var provider *ct.Provider
> > +
> > +	listRec(w, "ID", "Provider ID", "Provider Name")
> > +	for _, j := range resources {
> > +		provider, err = client.GetProvider(j.ProviderID)
>
> The new code has the potential to ignore errors (error, then success would
> hide the issue). I'd suggest just returning the error right away.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/flynn/flynn/pull/1137/files#r25658959>.
>


-- 
Patrick Collins